src/onegov/org/models/search.py

```diff
@@ -163,3 +169,241 @@ def suggestions(self) -> tuple[str, ...]:
         return tuple(self.request.app.es_suggestions_by_request(
             self.request, self.query
         ))
+
+
+def locale_mapping(locale: str) -> str:
+    mapping = {'de_CH': 'german', 'fr_CH': 'french', 'it_CH': 'italian',
+               'rm_CH': 'english'}
+    return mapping.get(locale, 'english')
+
+
+class SearchPostgres(Pagination[_M]):
+    """
+    Implements searching in postgres db based on the gin index
+    """
+    results_per_page = 10
+    max_query_length = 100
+
+    def __init__(self, request: OrgRequest, query: str, page: int):
+        self.request = request
+        self.query = query
+        self.page = page  # page index
+
+        self.number_of_docs = 0
+        self.number_of_results = 0
+
+        self.search_models = {
+            model for base in self.request.app.session_manager.bases
+            for model in searchable_sqlalchemy_models(base)}
```

Comment:
This is still not really correct for performing as few non-overlapping queries as possible. But if we go with a separate table for the search index, this should simplify away to some degree.